### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by rabayarabu**
- Add a column species of type string to animals table.
- Insert data.
- Inside a transaction update the animals table by setting the species column to unspecified.
- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Inside a transaction, I updated the table by setting the species column to unspecified and rollbacked the changes.
- Inside a transaction, I updated the table by setting the species column to 'digimon' for all animals that have a name ending in 'mon', and to 'pokemon' for all animals that don't have species already set, and I committed the changes.
- Inside a transaction, I deleted all records in the table, and then rolled back the transaction and below is the screenshot.
- Inside a transaction, I deleted all animals born after Jan 1st, 2022 and created a savepoint for the transaction, I updated all animals' weight to be their weight multiplied by -1, I rollbacked to the savepoint, I updated all animals' weights that are negative to be their weight multiplied by -1 and committed all transactions, and below are the screenshots

#Some analytical questions answers:

1. How many animals are there?
2. How many animals have never tried to escape?
3. What is the average weight of animals?
4. Who escapes the most, neutered or not neutered animals?
5. What is the minimum and maximum weight of each type of animal?
6. What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Here, are screenshots of my actions:

Screenshot 2023-07-19 113738
Screenshot 2023-07-19 114537
Screenshot 2023-07-19 114904
Screenshot 2023-07-19 114939
Screenshot 2023-07-19 115013
Screenshot 2023-07-19 115048
Screenshot 2023-07-19 115145
Screenshot 2023-07-19 115225
Screenshot 2023-07-19 115301
Screenshot 2023-07-20 162748
Screenshot 2023-07-20 165416

Screenshot 2023-07-20 164750

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg>10.5;
...
+ BEGIN;
+ UPDATE animals SET species='unspecified';
+ ROLLBACK;
+
+--  Updating the animal species
+ BEGIN;
+ UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+ UPDATE animals SET species='pokemon' WHERE species IS NULL;
+ SELECT * FROM animals;
+COMMIT;
```
Comment (mikethreels) [inline queries.sql:23]: - [ ] according to the requirements you should check if the changes persist after the commit as well.

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg>10.5;
...
+
+-- Delete Records
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (mikethreels) [inline queries.sql]: - [ ] according to the requirements you should check if the records are deleted before the rollback.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg>10.5;
...
+SELECT * FROM animals;
+BEGIN
+DELETE FROM animals WHERE date_of_birth>'2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg=weight_kg*-1;
+UPDATE animals SET weight_kg=weight_kg*-1 WHERE weight_kg<0;
```
Comment (mikethreels) [inline queries.sql]: - [ ] According to the requirements you're supposed to rollback to the savepoint in between the update transactions

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Rabu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job altering the table

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @rabayarabu 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the required changes 👍🏽
- All query results are accurate 👏🏽
- Descriptive PR title and summary 👌🏽
- Screenshots of results are present in the PR 👍🏼


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — COMMENTED** [review]
Hi @rabayarabu 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

approved-accepted

## Highlights
- Correct data inserted into the database✔✔
- Queries are written correctly :heavy_check_mark: 
- Screenshots show correct results :heavy_check_mark: 
- Readme looks professional :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**